tests/test_output_path_builder.py

```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Iterator
+
+import pytest
+
+from embeddings.utils.utils import build_output_path
+
+root = "results/experiment_1"
+hub_repo_official = "hub-123"
+hub_repo_unofficial = "org/hub-123"
+
+
+@pytest.fixture(scope="module")
+def local_repo() -> Iterator[Path]:
+    with TemporaryDirectory() as temp_path:
+        local_repo = Path(temp_path).joinpath("local-123")
+        local_repo.mkdir()
+        yield local_repo
```

Comment:
if the scope range is module then it will be called only once; why do we use Iterator and yield the Path instead of simply returning it? I realize that the outcome will be the same